wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt

```diff
@@ -155,29 +153,23 @@ data class JavaTarget(
         location: Location,
         context: Context,
       ): Path {
-        val outDirectory = context.outDirectory
         val javaFile = JavaFile.builder(javaTypeName.packageName(), typeSpec)
           .addFileComment("\$L", WireCompiler.CODE_GENERATED_BY_WIRE)
           .addFileComment("\nSource: \$L in \$L", source, location.withPathOnly())
           .build()
-        val filePath = outDirectory /
-          javaFile.packageName.replace(".", "/") /
+        val filePath = javaFile.packageName.replace(".", "/").toPath() /
           "${javaTypeName.simpleName()}.java"
 
-        context.logger.artifactHandled(
```

Comment:
Is there a reason to change this? Why not leave the `logger` field on context? When you create the context for the protoc plugin invocation, you can provide a custom implementation of `WireLogger` if you need. It seems strange to mix these methods into the file writer -- writing generated code and logging seem like two separate concerns that don't need to be conjoined into the same interface.

In the other PR, it looked like your custom implementation was just ignoring this log call. So you could leave the `logger` field and set it to `NULL_LOGGER`.

Reply:
The `NULL_LOGGER` is for tests but we can do something similar for the protoc case